StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp10/LayoutRules/SA1516CSharp10UnitTests.cs

```diff
@@ -5,9 +5,138 @@
 
 namespace StyleCop.Analyzers.Test.CSharp10.LayoutRules
 {
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.Testing;
     using StyleCop.Analyzers.Test.CSharp9.LayoutRules;
+    using Xunit;
+    using static StyleCop.Analyzers.Test.Verifiers.StyleCopCodeFixVerifier<
+        StyleCop.Analyzers.LayoutRules.SA1516ElementsMustBeSeparatedByBlankLine,
+        StyleCop.Analyzers.LayoutRules.SA1516CodeFixProvider>;
 
     public class SA1516CSharp10UnitTests : SA1516CSharp9UnitTests
     {
+        private const string CorrectCode = @"extern alias corlib;
+
+using System;
+using System.Linq;
+using a = System.Collections.Generic;
+
+namespace Foo;
+
+public class Bar
+{
+    public string Test1;
+    public string Test2;
+    public string Test3;
+
+    public string TestProperty1 { get; set; }
+
+    public string TestProperty2 { get; set; }
+    /// <summary>
+    /// A summary.
+    /// </summary>
+    public string TestProperty3 { get; set; }
+
+    public string TestProperty4
+    {
+        get
+        {
+            return Test1;
+        }
+
+        set
+        {
+            Test1 = value;
+        }
+    }
+
+    public string FooValue, BarValue;
+
+    [Obsolete]
+    public enum TestEnum
+    {
+        Value1,
+        Value2
+    }
+}
+
+public enum Foobar
+{
+
+}
+";
+
+        /// <summary>
+        /// Verifies that SA1516 is not reported for code with correct blank lines.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        [WorkItem(3512, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3512")]
+        public async Task TestFileScopedNamespaceCorrectSpacingAsync()
```

Comment:
💡 This test (and the associated constant) can be removed. All code fix tests verify that the final fixed code contains no diagnostics, so this test isn't providing significant value over `TestFileScopedNamespaceWrongSpacingAsync` below.